src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -0,0 +1,1111 @@
+
+  // Passing a empty string to encode characters
+  encoder.Encode(input, output, CharExtract());
+  vector<vector<double>> expected = {
+    { 1.4054651081081644, 2, 2.8109302162163288, 0, 0 },
+    { 0, 2, 2.8109302162163288, 2.8109302162163288, 2.0986122886681100 },
+    { 1.4054651081081644, 1, 0, 1.4054651081081644, 0 }
+  };
+  for (size_t i = 0; i < expected.size(); i++)
+    for (size_t j = 0; j < expected[i].size(); j++)
+      BOOST_REQUIRE_CLOSE(expected[i][j], output[i][j], 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using binary weighting scheme
+ * for tf and smoothidf as true.
+ */
+BOOST_AUTO_TEST_CASE(BinarySmoothIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::BINARY, true);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.2876820724517808, 1, 1.2876820724517808, 0, 0 },
+    { 0, 1, 1.2876820724517808, 1.2876820724517808, 1.6931471805599454 },
+    { 1.2876820724517808, 1, 0, 1.2876820724517808, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters to store results in vector, using binary
+ * weighting scheme for tf and smoothidf as true.
+ */
+BOOST_AUTO_TEST_CASE(VectorBnarySmoothIdfEncodingIndividualCharactersTest)
+{
+  std::vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  vector<vector<double>> output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::BINARY, true);
+
+  // Passing a empty string to encode characters
+  encoder.Encode(input, output, CharExtract());
+  vector<vector<double>> expected = {
+    { 1.2876820724517808, 1, 1.2876820724517808, 0, 0 },
+    { 0, 1, 1.2876820724517808, 1.2876820724517808, 1.6931471805599454 },
+    { 1.2876820724517808, 1, 0, 1.2876820724517808, 0 }
+  };
+  for (size_t i = 0; i < expected.size(); i++)
+    for (size_t j = 0; j < expected[i].size(); j++)
+      BOOST_REQUIRE_CLOSE(expected[i][j], output[i][j], 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using binary
+ * as weighting scheme and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(BinaryTfIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::BINARY, false);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.4054651081081644, 1, 1.4054651081081644, 0, 0 },
+    { 0, 1, 1.4054651081081644, 1.4054651081081644, 2.0986122886681100 },
+    { 1.4054651081081644, 1, 0, 1.4054651081081644, 0 }
+  };
```

Comment:
I wonder how did you get these values? `scikit-learn` doesn't support the binary mode.

Reply:
I test it manually for a smaller input and then when got correct values I just escalated for bigger values and saved it in a csv file

Reply:
Also there is an option to use binary mode , ` binary : boolean (default=False)
If True, all non-zero term counts are set to 1. This does not mean outputs will have only 0/1 values, only that the tf term in tf-idf is binary. (Set idf and normalization to False to get 0/1 outputs.) ` , refered from sklearn documentation.

Reply:
> Also there is an option to use binary mode

Ok, I'll try to reproduce.